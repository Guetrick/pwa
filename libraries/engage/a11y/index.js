/** @module a11y */

export { default as Navigation } from './components/Navigation';
export { default as Section } from './components/Section';
