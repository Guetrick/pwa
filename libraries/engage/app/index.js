import { APP_DID_START, APP_WILL_START, WILL_REGISTER_LINK_EVENTS, DID_REGISTER_LINK_EVENTS } from '@shopgate/pwa-common/constants/ActionTypes';

// CONSTANTS
export { APP_DID_START, APP_WILL_START, WILL_REGISTER_LINK_EVENTS, DID_REGISTER_LINK_EVENTS };

// ACTIONS
export { default as handleDeepLink } from '@shopgate/pwa-common/actions/app/handleDeepLink';
export { default as handleLink } from '@shopgate/pwa-common/actions/app/handleLink';
export { default as handlePushNotification } from '@shopgate/pwa-common/actions/app/handlePushNotification';
export { default as registerLinkEvents } from '@shopgate/pwa-common/actions/app/registerLinkEvents';

// STREAMS
export * from '@shopgate/pwa-common/streams/app';
