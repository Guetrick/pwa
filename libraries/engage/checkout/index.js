/** @module checkout */

// ACTIONS
export { default as fetchCheckoutUrl } from '@shopgate/pwa-common-commerce/checkout/actions/fetchCheckoutUrl';

// CONSTANTS
export * from '@shopgate/pwa-common-commerce/checkout/constants/index';
export * from '@shopgate/pwa-common-commerce/checkout/constants/Pipelines';

// SELECTORS
export * from '@shopgate/pwa-common-commerce/checkout/selectors';
