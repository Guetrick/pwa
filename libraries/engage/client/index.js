// ACTIONS
export { default as fetchClientInformation } from '@shopgate/pwa-common/actions/client/fetchClientInformation';
