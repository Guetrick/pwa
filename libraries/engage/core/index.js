/** @module core */

// Classes
export { default as AppCommand } from '@shopgate/pwa-core/classes/AppCommand';
export { default as GetAppPermissionsRequest } from '@shopgate/pwa-core/classes/AppPermissionsRequest/GetAppPermissionsRequest';
export { default as RequestAppPermissionsRequest } from '@shopgate/pwa-core/classes/AppPermissionsRequest/RequestAppPermissionsRequest';
export { default as BrightnessRequest } from '@shopgate/pwa-core/classes/BrightnessRequest';
export { default as Conditioner } from '@shopgate/pwa-core/classes/Conditioner';
export { default as DataRequest } from '@shopgate/pwa-core/classes/DataRequest';
export { default as DevServerBridge } from '@shopgate/pwa-core/classes/DevServerBridge';
export { default as errorManager, emitter } from '@shopgate/pwa-core/classes/ErrorManager';
export { default as event } from '@shopgate/pwa-core/classes/Event';
export { default as HttpRequest } from '@shopgate/pwa-core/classes/HttpRequest';
export { default as PipelineRequest } from '@shopgate/pwa-core/classes/PipelineRequest';
export { default as ScannerManager } from '@shopgate/pwa-core/classes/ScannerManager';
export { default as WebStorageRequest } from '@shopgate/pwa-core/classes/WebStorageRequest';

// Commands
export { default as analyticsSetCustomValues } from '@shopgate/pwa-core/commands/analyticsSetCustomValues';
export * from '@shopgate/pwa-core/commands/appPermissions';
export * from '@shopgate/pwa-core/commands/brightness';
export { default as broadcastEvent } from '@shopgate/pwa-core/commands/broadcastEvent';
export { default as cleanTab, cleanTabCmd } from '@shopgate/pwa-core/commands/cleanTab';
export { default as closeInAppBrowser } from '@shopgate/pwa-core/commands/closeInAppBrowser';
export { default as flushTab } from '@shopgate/pwa-core/commands/flushTab';
export { default as hideMenuBar } from '@shopgate/pwa-core/commands/hideMenuBar';
export { default as hideNavigationBar } from '@shopgate/pwa-core/commands/hideNavigationBar';
export { default as hideSplashScreen } from '@shopgate/pwa-core/commands/hideSplashScreen';
export { default as onload } from '@shopgate/pwa-core/commands/onload';
export { default as openPage } from '@shopgate/pwa-core/commands/openPage';
export { default as openPageExtern } from '@shopgate/pwa-core/commands/openPageExtern';
export {
  default as performCommandsAfterDelay,
  performCommandsAfterDelayCmd,
} from '@shopgate/pwa-core/commands/performCommandsAfterDelay';
export * from '@shopgate/pwa-core/commands/plotProjects';
export { default as popTabToRoot, popTabToRootCmd } from '@shopgate/pwa-core/commands/popTabToRoot';
export { default as registerEvents } from '@shopgate/pwa-core/commands/registerEvents';
export * from '@shopgate/pwa-core/commands/scanner';
export { default as setCookie } from '@shopgate/pwa-core/commands/setCookie';
export { default as setDebugLoggingEnabled } from '@shopgate/pwa-core/commands/setDebugLoggingEnabled';
export { default as setScrollingEnabled } from '@shopgate/pwa-core/commands/setScrollingEnabled';
export { default as showNavigationBar } from '@shopgate/pwa-core/commands/showNavigationBar';
export { default as showTab } from '@shopgate/pwa-core/commands/showTab';
export * from '@shopgate/pwa-core/commands/unifiedTracking';
export * from '@shopgate/pwa-core/commands/webStorage';

// Constants
export * from '@shopgate/pwa-core/constants/AppEvents';
export * from '@shopgate/pwa-core/constants/AppPermissions';
export * from '@shopgate/pwa-core/constants/ErrorHandleTypes';
export * from '@shopgate/pwa-core/constants/Pipeline';
export * from '@shopgate/pwa-core/constants/ProcessTypes';
export * from '@shopgate/pwa-core/constants/Scanner';

// Emitters
export { default as UIEvents } from '@shopgate/pwa-core/emitters/ui';

// Helpers
export * from '@shopgate/pwa-core/helpers';
export { default as logGroup } from '@shopgate/pwa-core/helpers/logGroup';
export * from '@shopgate/pwa-core/helpers/version';
