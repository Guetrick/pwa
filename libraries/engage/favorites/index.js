/** @module favorites */

// ACTIONS
export { default as fetchFavorites } from '@shopgate/pwa-common-commerce/favorites/actions/fetchFavorites';
export { requestForceFavoritesBufferClear as toggleFavorites } from '@shopgate/pwa-common-commerce/favorites/actions/toggleFavorites';

// CONSTANTS
export * from '@shopgate/pwa-common-commerce/favorites/constants/index';
export * from '@shopgate/pwa-common-commerce/favorites/constants/Pipelines';
export * from '@shopgate/pwa-common-commerce/favorites/constants/Portals';

// SELECTORS
export * from '@shopgate/pwa-common-commerce/favorites/selectors';

// STREAMS
export * from '@shopgate/pwa-common-commerce/favorites/streams';
