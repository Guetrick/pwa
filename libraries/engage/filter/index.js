/** @module filter */

// ACTIONS
export { default as fetchFilters } from '@shopgate/pwa-common-commerce/filter/actions/fetchFilters';

// CONSTANTS
export * from '@shopgate/pwa-common-commerce/filter/constants/index';
export * from '@shopgate/pwa-common-commerce/filter/constants/Pipelines';
export * from '@shopgate/pwa-common-commerce/filter/constants/Portals';

// SELECTORS
export * from '@shopgate/pwa-common-commerce/filter/selectors';

// STREAMS
export * from '@shopgate/pwa-common-commerce/filter/streams';
