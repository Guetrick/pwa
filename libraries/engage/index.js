/** @module root */
import { themeConfig } from '@shopgate/pwa-common/helpers/config';

export { themeConfig };
export { default as appConfig } from '@shopgate/pwa-common/helpers/config';
