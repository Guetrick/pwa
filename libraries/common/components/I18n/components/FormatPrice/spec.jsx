import React from 'react';
import { mount } from 'enzyme';
import { i18n } from '@shopgate/engage/core';
import I18n from '../../index';

jest.unmock('@shopgate/engage/core/helpers/i18n');

describe('<FormatPrice />', () => {
  const locales = {
    greeting: 'Hello {price}',
  };
  const price = 1234.56;
  const formattedPrice = 'â‚¬1,234.56';
  const currency = 'EUR';
  const lang = 'en-US';

  i18n.init({
    locales,
    lang,
  });

  describe('Given the component was mounted to the DOM', () => {
    let renderedElement;
    it('should match snapshot', () => {
      renderedElement = mount((
        <I18n.Provider>
          <div>
            <span className="only-price">
              <I18n.Price price={price} currency={currency} />
            </span>
            <span className="text-with-price">
              <I18n.Text string="greeting">
                <I18n.Price forKey="price" price={price} currency={currency} />
              </I18n.Text>
            </span>
          </div>
        </I18n.Provider>
      ));
      expect(renderedElement).toMatchSnapshot();
    });

    it('should render formatted price', () => {
      const text = renderedElement.find('.only-price').text();
      expect(text).toBe(formattedPrice);
    });

    it('should render within translated text', () => {
      const text = renderedElement.find('.text-with-price').text();
      expect(text).toBe(`Hello ${formattedPrice}`);
    });
  });
});
