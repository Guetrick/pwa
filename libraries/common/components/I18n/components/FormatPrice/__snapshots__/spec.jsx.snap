// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormatPrice /> Given the component was mounted to the DOM should match snapshot 1`] = `
<I18nProvider>
  <div>
    <span
      className="only-price"
    >
      <FormatPrice
        className=""
        currency="EUR"
        fractions={true}
        price={1234.56}
      >
        <span
          className=""
        >
          €1,234.56
        </span>
      </FormatPrice>
    </span>
    <span
      className="text-with-price"
    >
      <Translate
        className=""
        params={Object {}}
        role={null}
        string="greeting"
      >
        <span
          className=""
          role={null}
        >
          Hello 
          <FormatPrice
            className=""
            currency="EUR"
            forKey="price"
            fractions={true}
            key=".0"
            price={1234.56}
          >
            <span
              className=""
            >
              €1,234.56
            </span>
          </FormatPrice>
        </span>
      </Translate>
    </span>
  </div>
</I18nProvider>
`;
